go/apps/mixnet/proxy.go

```diff
@@ -18,39 +18,60 @@ import (
 	"encoding/binary"
 	"errors"
 	"io"
+	"net"
+	"time"
 
+	"github.com/golang/glog"
 	"github.com/jlmucb/cloudproxy/go/tao"
 )
 
 // ProxyContext stores the runtime environment for a mixnet proxy. A mixnet
 // proxy connects to a mixnet router on behalf of a client's application.
 type ProxyContext struct {
-	domain *tao.Domain // Policy guard and public key.
-	id     uint64      // Next serial identifier that will assigned to a new connection.
+	domain   *tao.Domain    // Policy guard and public key.
+	listener *SocksListener // SOCKS5 server for listening to clients.
```

Comment:
[optional]: make this a net.Listener for generality, since that allows us to swap in other protocols easily in the future. This would mean taking a net.Listener in NewProxyContext and constructing a SocksListener to pass into it in our binaries and tests.